docker-compose.yml

```diff
@@ -1,22 +1,117 @@
 version: '3.2'
 services:
   geth:
-    image: "matterlabs/geth:latest"
+    image: "ethereum/client-go:v1.13.5"
     ports:
-      - "8545:8545"
-      - "8546:8546"
+      - 8551:8551
+      - 8545:8545
+      - 8546:8546
     volumes:
-    - type: bind
-      source: ./volumes/geth
-      target: /var/lib/geth/data
+      - ./volumes/geth:/var/lib/geth/data
+      - ./docker/geth/:/geth/:ro
+    command:
+      - --networkid=9
+      - --datadir=/var/lib/geth/data
+      - --http
+      - --http.api=engine,eth,web3,personal,net,debug
+      - --http.addr=0.0.0.0
+      - --http.corsdomain=*
+      - --http.vhosts=*
+      - --ws
+      - --ws.addr=0.0.0.0
+      - --ws.port=8546
+      - --ws.origins=*
+      - --nodiscover
+      - --authrpc.addr=0.0.0.0
+      - --authrpc.vhosts=*
+      - --authrpc.jwtsecret=/var/lib/geth/data/jwtsecret
+      - --allow-insecure-unlock
+      - --unlock=0x8a91dc2d28b689474298d91899f0c1baf62cb85b
+      - --password=/var/lib/geth/data/password.sec
+      - --syncmode=full
+    depends_on:
+      beacon:
+        condition: service_started
+  beacon:
+    image: "gcr.io/prylabs-dev/prysm/beacon-chain:v4.1.1"
+    command:
+      - --datadir=/consensus/beacon/
+      - --min-sync-peers=0
+      - --genesis-state=/consensus/genesis.ssz
+      - --bootstrap-node=
+      - --interop-eth1data-votes
+      - --chain-config-file=/consensus/config.yml
+      - --contract-deployment-block=0
+      - --chain-id=9
+      - --rpc-host=0.0.0.0
+      - --grpc-gateway-host=0.0.0.0
+      - --execution-endpoint=http://geth:8551
+      - --accept-terms-of-use
+      - --jwt-secret=/execution/jwtsecret
+      - --suggested-fee-recipient=0x8a91dc2d28b689474298d91899f0c1baf62cb85b
+      - --minimum-peers-per-subnet=0
+      - --enable-debug-rpc-endpoints
+    ports:
+      - 4000:4000
+      - 3500:3500
+      - 8080:8080
+      - 6060:6060
+      - 9090:9090
+    volumes:
+      - ./volumes/prysm:/consensus
+      - ./volumes/geth:/execution
+  validator:
+    image: "gcr.io/prylabs-dev/prysm/validator:v4.1.1"
+    command:
+      - --beacon-rpc-provider=beacon:4000
+      - --datadir=/consensus/validatordata
+      - --accept-terms-of-use
+      - --interop-num-validators=64
+      - --interop-start-index=0
+      - --chain-config-file=/consensus/config.yml
+    depends_on:
+      beacon:
+        condition: service_started
+    volumes:
+      - ./volumes/prysm:/consensus
   postgres:
     image: "postgres:14"
     ports:
-      - "5432:5432"
+      - 5432:5432
     volumes:
       - type: bind
         source: ./volumes/postgres
         target: /var/lib/postgresql/data
     environment:
       - POSTGRES_HOST_AUTH_METHOD=trust
-
+  
+  # This is specific to runner
+  zk:
+    image: "matterlabs/zk-environment:latest2.0-lightweight"
+    depends_on:
+      - geth
+      - postgres
+      - beacon
```

Comment:
we have additional docker-compose  for runner

Reply:
why?

Reply:
they need to contain the same `geth` `postgres` `validator` `beacon` services and on top of that runner has to contain `zk`. I am not going to copypaste  the first 4 into two different configs, too error-prone. We can just enable `zk` with `profiles` when needed.